src/routes/pdf-fetcher/containers/PdfFetcherPage.js

```diff
@@ -31,20 +31,18 @@ class PdfFetcherPage extends React.Component {
   }
 
   componentWillMount () {
-    const hierarchy = new Hierarchy(this.props.path)
-
-    const error = hierarchy.build(this.props.pages)
-    if (error) {
-      return <Error error={error}/>
+    const category = this.props.categories.getCategoryByUrl(this.props.fetchUrl)
+    if (category) {
+      this.fetchUrl(category)
+    } else {
+      // todo display an error
     }
```

Comment:
You can remove this else block, because if we don't call fetchUrl() our state will always be {pdf: '', loading: false}, and therefore the render() method will return an Error-component.

Reply:
done